### PR TITLE
docs: Update inline docs on action/upgrade.go

### PR DESCRIPTION
**Pull request by technosophos**
See #7807 for an example of how the absence of docs confused a user.

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

**Comment by technosophos** [comment]
Also tried to clarify vis a vis #7811

File: pkg/action/upgrade.go [hunk 1]
```diff
@@ -42,27 +42,53 @@ type Upgrade struct {
...
+	MaxHistory int
+	// Atomic, if true, will roll back on failure.
+	Atomic bool
+	// CleanupOnFail will, if true, cause the upgrade to delete newly-created resources on a failed update.
+	CleanupOnFail bool
+	// SubNotes determines whether sub-notes are rendered in the chart.
+	SubNotes bool
+	// Description is the description of this operation
+	Description  string
+	PostRenderer postrender.PostRenderer
```
Comment (hickeyma) [inline pkg/action/upgrade.go]: Might be worth adding a description for the post renderer also?

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for adding the descriptions @technosophos. Looking good, just missing a description.

File: pkg/action/upgrade.go (hunk 1 above)
Comment (technosophos) [reply pkg/action/upgrade.go:94]: Can I please get a re-review on this?

**Review by jdolitsky — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @technosophos

**Pull request merged**